Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -117,6 +120,25 @@ QueryData CreateQueryData(absl::string_view str) {
   return CreateQueryData(Query(str));
 }
 
+bool operator==(const WatchChange& lhs, const WatchChange& rhs) {
+  if (lhs.type() != rhs.type()) {
+    return false;
+  }
+  switch (lhs.type()) {
```

Comment:
There's already a helper function like this [here](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Example/Tests/Remote/FSTSerializerBetaTests.mm#L116-L136). I'd suggest just copying it over.